colossalai/zero/sharded_model/sharded_model_v2.py

```diff
@@ -59,8 +61,16 @@ def __init__(self,
                 if self.shard_param:
                     self.shard_strategy.shard([param.col_attr.data])
 
+        # Init Memory Statistics Collector
+        self._use_memory_tracer = use_memory_tracer
+        if self._use_memory_tracer:
+            self._memstats_collector = MemStatsCollector()
```

Comment:
This collector is designed only for `ShardedModel` or for all models? Should we manage it in `engine`?

Reply:
Now just used for ShardedModel. It highly depends on the zerohook.